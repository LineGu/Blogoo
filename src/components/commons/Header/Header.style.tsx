import styled from '@emotion/styled';

export const HeaderWrapper = styled.header`
    display: flex;
    align-items: center;
    width: 100vw;
    height: 4rem;
`;

export const Title = styled.a`
    font-family: 'Pretendard-Bold';
    text-decoration: none;
    font-size: 2rem;
    font-weight: 800;
    color: black;

    &:hover {
        cursor: pointer;
    }
`;
