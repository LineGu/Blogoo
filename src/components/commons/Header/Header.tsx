import React, { ReactElement } from 'react';
import Link from 'next/link';
import PATH from '@constants/path';
import Grid from '@components/commons/Grid';
import * as S from './Header.style';

const ASSETS = {
    TITLE: `Goo's Blog`,
};

function Header(): ReactElement {
    return (
        <S.HeaderWrapper>
            <Grid>
                <Link href={PATH.HOME}>
                    <S.Title>{ASSETS.TITLE}</S.Title>
                </Link>
            </Grid>
        </S.HeaderWrapper>
    );
}

export default Header;
