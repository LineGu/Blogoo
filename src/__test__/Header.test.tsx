import React from 'react';
import { render } from './test-utils';
import Header from '@components/commons/Header';

describe('Header Component', () => {
    it('match snapshot', () => {
        const utils = render(<Header />);
        expect(utils.container).toMatchSnapshot();
    });
    it('has correct name', () => {
        const utils = render(<Header />);

        expect(utils.getByText(`Goo's Blog`)).toBeInTheDocument();
    });
});
