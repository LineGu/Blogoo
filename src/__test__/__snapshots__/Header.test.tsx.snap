// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Component match snapshot 1`] = `
<div>
  <header
    class="css-jc1se7"
  >
    <div
      class="css-1xuqsws"
    >
      <a
        class="css-ids50t"
      >
        Goo's Blog
      </a>
    </div>
  </header>
</div>
`;
